cloudformation.yml

```diff
@@ -1,65 +1,80 @@
 #adapted from : https://github.com/Integralist/CloudFormation/blob/master/EC2.yml
 
-AWSTemplateFormatVersion: '2010-09-09' 
+AWSTemplateFormatVersion: 2010-09-09
 
-# Paramaters are referenced using the Ref property
-# e.g. Ref: "KeyName"
+# Parameters are referenced using the Ref property
+# e.g. Ref: KeyNameParameter
 #
 # Parameters can be assigned a default value, but if not then the value
 # needs to be provided when uploading the CloudFormation.
 # If uploading via CLI then you'll provide a --parameters flag
 # If uploading via the AWS console, a GUI will be provided
 Parameters:
-  KeyName:
+  KeyNameParameter:
     Description: "Optional: Name of an existing EC2 KeyPair to enable SSH access to the instance"
     Type: String
     Default: ""
-  InstanceTypeParameter: 
+  SubnetIdParameter:
+    Description: "Enter the existing subnet ID in which the EC2 instance will be created (ex: sg-342109b9)"
     Type: String
-    Description: "Enter an EC2 instance type."
+    Default: ""
+    ConstraintDescription: "[A-Za-z]+-[A-Za-z]+"
+  VpcIdParameter:
```

Comment:
Could you move out the conflicting part so I can integrate your work within the existing template.